File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:

        mp={0:1}

        count=0;
        sum=0;

        for i in range(len(nums)):
            sum+=nums[i]
            p=sum-k
            if p in mp:
                count+=mp[p]

            mp[sum]=mp.get(sum,0)+1


        return count        

        

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {

        unordered_map<int,int>ans;
        ans[0]=1;

        int sum=0;
        int count=0;

        for(int i=0;i<nums.size();i++){
            sum+=nums[i];

            int x=sum-k;
            if(ans.find(x)!=ans.end()){
                count+=ans[sum-k];
            }

            ans[sum]++;
        }

       return count; 
    }
};